pkg/services/blockchain/eth/resolver.go

```diff
@@ -148,9 +245,206 @@ func (r *Resolver) Resolve(
 		}
 	}
 
+	signature := ""
+	if opts.Signature != "" {
+		if r.walletKey == "" {
+			return services.IdentityState{},
+				errors.New("no wallet key found for generating signature")
+		}
+		primaryType := services.IdentityStateType
+		if opts.GistRoot != nil {
```

Comment:
what if both options will persist in query?

Reply:
Only 1 of them is needed. Because in eip712 signature we only sign 1 message GlobalState or IdentityState. So if we pass `GistRoot` we will force the message as GlobalState.

Reply:
With signature option we are only signing 1 message. GlobalState message (for user) or IdentityState message (for issuer). If we send GistRoot will be prioritized and be signed GlobalState message.